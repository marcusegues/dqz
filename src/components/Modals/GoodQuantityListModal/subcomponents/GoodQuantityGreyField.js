// @flow
import React from 'react';
import { translate } from 'react-i18next';
import type { ComponentType } from 'react';
// $FlowFixMe
import { Text, View } from 'react-native';
import { CardHeaderSubText } from '../../../QuestionAnswer/Cards/subcomponents/CardHeaderSubText';
import { moderateScale, verticalScale } from '../../../../styles/Scaling';
import { MAIN_RED } from '../../../../styles/colors';
import { RedPlusIcon } from './RedPlusIcon';

const ownStyles = {
  container: {
    flex: 1,
    width: '90%',
    flexDirection: 'column',
    justifyContent: 'space-between',
    alignItems: 'center',
    backgroundColor: '#F5F5F5',
    marginBottom: 0,
  },
  topText: {
    alignSelf: 'flex-start',
    paddingTop: verticalScale(16),
    paddingBottom: verticalScale(35),
  },
  plusIconText: {
    fontSize: moderateScale(14),
    color: MAIN_RED,
    fontFamily: 'roboto_medium',
    marginTop: verticalScale(10),
    marginBottom: verticalScale(30),
  },
};

type GoodQuantityGreyFieldProps = {
  topText: string,
  plusIconText: string,
  onPress: () => void,
};

const GoodQuantityGreyFieldInner = ({
  topText,
  plusIconText,
  onPress,
}: GoodQuantityGreyFieldProps) => (
  <View style={ownStyles.container}>
    <CardHeaderSubText text={topText} style={ownStyles.topText} />
    <View style={{ alignItems: 'center' }}>
      <RedPlusIcon onPress={() => onPress()} />
      <Text style={ownStyles.plusIconText}>{plusIconText.toUpperCase()}</Text>
    </View>
  </View>
);

export const GoodQuantityGreyField = (translate([
  'mainCategories',
  'categories',
])(GoodQuantityGreyFieldInner): ComponentType<GoodQuantityGreyFieldProps>);
