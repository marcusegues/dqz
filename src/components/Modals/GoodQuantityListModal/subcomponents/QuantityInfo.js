// @flow
import React from 'react';
import { translate } from 'react-i18next';
import type { ComponentType } from 'react';
// $FlowFixMe
import { View } from 'react-native';
import { CardHeaderText } from '../../../QuestionAnswer/cards/subcomponents/CardHeaderText';
import { CardHeaderSubText } from '../../../QuestionAnswer/cards/subcomponents/CardHeaderSubText';
import { QuantityIcon } from '../../../General Components/GreyBox/configured/QuantityIcon';
import { verticalScale } from '../../../../styles/Scaling';
import { CategoriesInfo } from '../../../../model/constants';
import type { MainCategory } from '../../../../types/reducers/appReducer';
import type { Category } from '../../../../model/types/basketPeopleAmountsTypes';
import type { TFunction } from '../../../../types/generalTypes';

type QuantityInfoProps = {
  mainCategory: MainCategory,
  category: Category,
  totalQuantity: number,
};

const QuantityInfoInner = ({
  mainCategory,
  category,
  totalQuantity,
  t,
}: QuantityInfoProps & { t: TFunction }) => (
  <View
    style={{
      flexDirection: 'column',
      alignSelf: 'flex-start',
      alignItems: 'flex-start',
      margin: verticalScale(16),
    }}
  >
    <CardHeaderText text={t(`categories:${category}`)} />
    <CardHeaderSubText text={t(`mainCategories:${mainCategory}`)} />
    <View style={{ marginTop: 10 }}>
      <QuantityIcon
        quantity={totalQuantity}
        unit={CategoriesInfo.getIn([category, 'unit'], '')}
      />
    </View>
  </View>
);

export const QuantityInfo = (translate(['mainCategories', 'categories'])(
  QuantityInfoInner
): ComponentType<QuantityInfoProps>);
