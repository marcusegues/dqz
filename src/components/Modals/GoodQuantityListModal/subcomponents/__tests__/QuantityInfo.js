// @flow
import React from 'react';
// $FlowFixMe
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { GoodQuantityGreyField } from '../GoodQuantityGreyField';
import { QuantityInfo } from '../QuantityInfo';

jest.mock('react-i18next', () => ({
  translate: () => Component => props => <Component t={() => ''} {...props} />,
}));

const mockFunc = jest.fn();

describe('QuantityInfo', () => {
  test('QuantityInfo with `category` and `totalQuantity` renders correctly', () => {
    const tree = renderer
      .create(<QuantityInfo category="Meat" totalQuantity={4} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  // test('QuantityInfo with `category` and `totalQuantity` renders correctly', () => {
  //   const tree = renderer
  //     // $FlowFixMe
  //     .create(<QuantityInfo />)
  //     .toJSON();
  //   expect(tree).toMatchSnapshot();
  // });
});
