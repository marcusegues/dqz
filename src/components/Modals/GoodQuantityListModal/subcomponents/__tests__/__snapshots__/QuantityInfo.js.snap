// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuantityInfo QuantityInfo with \`category\` and \`totalQuantity\` renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-start",
      "alignSelf": "flex-start",
      "flexDirection": "column",
      "margin": 31.388235294117646,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#141414",
        "fontFamily": "roboto_regular",
        "fontSize": 24,
        "lineHeight": 62.77647058823529,
      }
    }
  >
    
  </Text>
  <View
    style={
      Object {
        "marginTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#E0E0E1",
          "borderRadius": 3,
          "paddingHorizontal": 8,
          "paddingVertical": 2,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "black",
            "fontFamily": "roboto_regular",
            "fontSize": 14,
          }
        }
      >
        4.000 
      </Text>
    </View>
  </View>
</View>
`;
